utils/config_update_utils.py

```diff
@@ -0,0 +1,111 @@
+import os
+import logging
+from jose import jwt
+import arrow
+import requests
+
+
+CONFIG_REPO = os.getenv("CONFIG_REPO")
+WORKFLOW = os.getenv("CONFIG_UDPATES_WORKFLOW")
+GH_APP_ID = os.getenv("CONFIG_UPDATES_GH_APP_ID")
+GH_APP_INSTALLATION_ID = os.getenv("CONFIG_UPDATES_GH_APP_INSTALLATION_ID")
+GH_APP_PRIVATE_KEY = os.getenv("CONFIG_UPDATES_GH_APP_PRIVATE_KEY")
+
+
+def generate_jwt():
+    if not GH_APP_ID or not GH_APP_PRIVATE_KEY:
+        raise ValueError("GH app credentials not set in env")
+
+    now = int(arrow.utcnow().timestamp())
+    payload = {
+        "iat": now,
+        "exp": now + 600,  # 10 min
+        "iss": GH_APP_ID,
+    }
+
+    encoded_jwt = jwt.encode(payload, GH_APP_PRIVATE_KEY, algorithm="RS256")
+    return encoded_jwt
+
+
+def get_installation_access_token():
+    if not GH_APP_INSTALLATION_ID:
+        raise ValueError("GH app installation ID not set in env")
+
+    jwt_token = generate_jwt()
+
+    headers = {
+        "Authorization": f"Bearer {jwt_token}",
+        "Accept": "application/vnd.github+json"
+    }
+
+    url = f"https://api.github.com/app/installations/{GH_APP_INSTALLATION_ID}/access_tokens"
+    response = requests.post(url, headers=headers)
+    response.raise_for_status()
+
+    return response.json()["token"]
+
+
+def trigger_config_update_workflow(inputs):
+    try:
+        token = get_installation_access_token()
+    except ValueError as e:
+        logging.warning(f"Failed to get installation access token: {e}")
+        return 500
+
+    inputs['token'] = token
```

Comment:
I would suggest adding the user in the admin dashboard (e.g. john.doe@cityofatlantis.com) as one of the inputs and have it be reflected in the PR. Since all these PRs will come from the same GitHub user, passing the real user though this way will allow us to maintain an audit trail of who made the appropriate changes.

Reply:
I am not sure how to determine what user is using the dashboard

Reply:
I would assume that you would get it from the JWT
https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-the-id-token.html

which we read here:
https://github.com/e-mission/op-admin-dashboard/blob/6d24ccb9a97d54af3eb53c6382cd0e52365c009f/utils/cognito_utils.py#L75